### PR TITLE
update-test-infra script handles repos with no Gopkg.lock file

### PR DESCRIPTION
**Pull request by chaodaiG**
`dep ensure -update` command fails if no `Gopkg.lock` file is found, and fails`hack/update-test-infra.sh` script, this happens in repo like observability. Handle it so this script doesn't fail in this scenario

### CODE REVIEW

File: hack/update-test-infra.sh [hunk 1]
```diff
@@ -37,12 +37,16 @@ for repo in *; do
...
-  echo -e "\nCheck the PR created above, and make changes if necessary"
+  if [[ -f "Gopkg.lock" ]]; then
+    dep ensure -update github.com/knative/test-infra
+    ./hack/update-deps.sh
+    [[ -z "$(git diff)" ]] && continue
+    git commit -a -m "Update test-infra to the latest version"
+    git push -u origin ${branch}
+    echo -e "\nCheck the PR created above, and make changes if necessary"
+  else
+    echo -e "\nGopkg.lock not found, skip updating"
```
Comment (adrcunha) [inline hack/update-test-infra.sh]: If you find `go.mod`, update using `go get` and `go mod vendor`. This is the case for observability and client repos.

Comment (chaodaiG) [reply hack/update-test-infra.sh]: so do `go get` and `go mod vendor` instead of `dep ensure`, right?

Comment (adrcunha) [reply hack/update-test-infra.sh]: Yes.

```
GO111MODULE=on go get -u github.com/knative/test-infra/scripts
GO111MODULE=on go mod vendor
```

Comment (chaodaiG) [reply hack/update-test-infra.sh]: This command failed in `observability`:
```
GO111MODULE=on go get -u github.com/knative/test-infra/scripts
go: dmitri.shuralyov.com/html/belt@v0.0.0-20180602232347-f7d459c86be0: unrecognized import path "dmitri.shuralyov.com/html/belt" (https fetch: Get https://dmitri.shuralyov.com/html/belt?go-get=1: dial tcp 172.93.50.41:443: connect: no route to host)
go: dmitri.shuralyov.com/service/change@v0.0.0-20181023043359-a85b471d5412: unrecognized import path "dmitri.shuralyov.com/service/change" (https fetch: Get https://dmitri.shuralyov.com/service/change?go-get=1: dial tcp 172.93.50.41:443: connect: no route to host)
go: dmitri.shuralyov.com/app/changes@v0.0.0-20180602232624-0a106ad413e3: unrecognized import path "dmitri.shuralyov.com/app/changes" (https fetch: Get https://dmitri.shuralyov.com/app/changes?go-get=1: dial tcp 172.93.50.41:443: connect: no route to host)
go: dmitri.shuralyov.com/state@v0.0.0-20180228185332-28bcc343414c: unrecognized import path "dmitri.shuralyov.com/state" (https fetch: Get https://dmitri.shuralyov.com/state?go-get=1: dial tcp 172.93.50.41:443: connect: no route to host)
go: error loading module requirements
```

I will take a look at what' going on

Comment (chaodaiG) [reply hack/update-test-infra.sh]: it works now, will update the PR

Comment (chaodaiG) [reply hack/update-test-infra.sh]: Updated

**Comment by adrcunha** [comment]
Thanks!

/lgtm

**Pull request merged**